tests/granulate_utils/config_feeder/fixtures/base.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any, Callable, ContextManager, Dict, List, Tuple, TypeVar
+from unittest.mock import Mock, mock_open, patch
+
+from granulate_utils.config_feeder.client.bigdata import get_node_info
+from granulate_utils.config_feeder.core.models.node import NodeInfo
+
+T = TypeVar("T", bound="NodeMockBase")
+
+
+class NodeMockBase(ABC):
+    def __init__(self) -> None:
+        self._files: Dict[str, str] = {}
+        self._stdout: Dict[str, bytes | str] = {}
+        self._contexts: List[Tuple[ContextManager[Any], Callable[[Any], None] | None]] = []
+
+    @property
+    def node_info(self) -> NodeInfo:
+        node_info = get_node_info()
+        assert node_info is not None
+        return node_info
+
+    def mock_file(self: T, fname: str, content: str) -> T:
```

Comment:
```suggestion
    def mock_file(self: T, fname: str, content: str) -> NodeMockBase:
```

Isn't good enough, I understand there's inheritance, but mypy should take care of it